server/src/main/java/org/apache/druid/discovery/DiscoveryDruidNode.java

```diff
@@ -61,6 +73,28 @@ public DiscoveryDruidNode(
     }
   }
 
+  @JsonCreator
+  private static DiscoveryDruidNode fromJson(
+      @JsonProperty("druidNode") DruidNode druidNode,
+      @JsonProperty("nodeType") NodeRole nodeRole,
+      @JsonProperty("services") Map<String, Map<String, Object>> rawServices,
+      @JacksonInject ObjectMapper jsonMapper
+  )
+  {
+    Map<String, DruidService> services = new HashMap<>();
+    if (rawServices != null && !rawServices.isEmpty()) {
```

Comment:
nit: can use `org.apache.druid.utils.CollectionUtils.isNullOrEmpty` to check the `rawServices`

Reply:
Oh, that method cannot be used as `Map` is not a `Collection`.

Reply:
Oh, right. Maybe we can add some overridden methods to this util class in the future.